pkg/restore/restore.go

```diff
@@ -1229,12 +1257,18 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 			return warnings, errs
 		}
 		// Remove insubstantial metadata.
-		fromCluster, err = resetMetadataAndStatus(fromCluster)
+		fromCluster, err = resetMetadata(fromCluster)
 		if err != nil {
 			ctx.log.Infof("Error trying to reset metadata for %s: %v", kube.NamespaceAndName(obj), err)
 			warnings.Add(namespace, err)
 			return warnings, errs
 		}
+		fromCluster, err = resetStatus(fromCluster)
```

Comment:
As far as my understanding: since the status has been removed from "fromCluster" here, but the status in "obj" is not removed when shouldRestoreStatus = true, this will make equality.Semantic.DeepEqual(fromCluster, obj)@1279 returns false all the time, which changes the original logic. Am I wrong?

Reply:
This part of the code is doing the same thing, just split the two tasks...
But I believe you're right. In the scenario that you're restoring status (shouldRestoreStatus = true), and the resource already exists, as is, it's always triggering the update.
Let me try to fix it

Reply:
I just pushed a fix for this, let me know what you think